quine-endpoints/src/main/scala/com/thatdot/quine/routes/IngestRoutes.scala

```diff
@@ -145,7 +145,24 @@ sealed abstract class KafkaSecurityProtocol(val name: String)
 object KafkaSecurityProtocol {
   case object PlainText extends KafkaSecurityProtocol("PLAINTEXT")
   case object Ssl extends KafkaSecurityProtocol("SSL")
-  val values: Seq[KafkaSecurityProtocol] = Seq(PlainText, Ssl)
+  case object Sasl_Ssl extends KafkaSecurityProtocol("SASL_SSL")
+  val values: Seq[KafkaSecurityProtocol] = Seq(PlainText, Ssl, Sasl_Ssl)
+}
+
+@title("Kafka SASL Authentication")
+@docs(
+  "See [SASL authentication in the Kafka documentation](https://docs.confluent.io/4.1.3/kafka/authentication_sasl.html)."
+)
+sealed abstract class KafkaSaslAuthentication
+object KafkaSaslAuthentication {
+  @title("Kafka SASL PLAIN Authentication")
+  @docs(
+    "See [SASL PLAIN authentication in the Kafka documentation](https://docs.confluent.io/4.1.3/kafka/authentication_sasl.html#plain)."
+  )
+  final case class Plain(
+    jaasConfig: String = "org.apache.kafka.common.security.plain.PlainLoginModule required username='' password='';",
+    saslMechanism: String = "PLAIN"
```

Comment:
Perhaps the `sealed abstract class KafkaSaslAuthentication` could be `sealed abstract class KafkaSaslAuthentication(val saslMechanism: String)`, where this subclass could be case class `Plain(...) extends KafkaSaslAuthentication("PLAIN")`, so the end-user would have that as a field to specify when supplying this. I guess the end-user could pass something like  `{"type": "Plain", "username": "foo", "password": "bar"}`, the `"type"` field being the discriminator that the endpoints4s  uses to deserialize JSON object to the appropriate instance of `KafkaSaslAuthentication`. 
As-is, it seems there's nothing "PLAIN"-specific about this class, besides the default values - it could equivalently be made into the sole instance of `KafkaSaslAuthentication`, and the user could use whatever SASL mechnism they wanted by specifying these two strings?